### PR TITLE
Implement `--service-name` as swap-in replacement for `--service-id`

### PR DESCRIPTION
**Pull request by Integralist**
Fixes https://github.com/fastly/cli/issues/481

### CODE REVIEW

File: pkg/cmd/flags.go [hunk 1]
```diff
@@ -30,6 +31,14 @@ func (b Base) RegisterServiceIDFlag(dst *string) {
 	b.CmdClause.Flag("service-id", "Service ID (falls back to FASTLY_SERVICE_ID, then fastly.toml)").Short('s').StringVar(dst)
 }
 
+// RegisterServiceNameFlag defines a --service-name flag that will attempt to
+// acquire the Service ID associated with the given service name.
+//
+// See: cmd.OptionalServiceNameID.Parse()
+func (b Base) RegisterServiceNameFlag(action kingpin.Action, dst *string) {
+	b.CmdClause.Flag("service-name", "The name of the service").Action(action).StringVar(dst)
+}
```
Comment (Integralist) [inline pkg/cmd/flags.go:40]: Here's the new abstraction for creating a `--service-name` flag.

File: pkg/cmd/flags.go [hunk 2]
```diff
@@ -94,6 +103,26 @@ func (sv *OptionalServiceVersion) Parse(sid string, client api.Interface) (*fast
...
+		return serviceID, fmt.Errorf("error listing services: %w", err)
+	}
+	for _, s := range services {
+		if s.Name == sv.Value {
+			return s.ID, nil
+		}
+	}
+	return serviceID, errors.New("error matching service name with available services")
+}
+
```
Comment (Integralist) [inline pkg/cmd/flags.go:125]: Here is where we implement the logic for mapping a service name to an ID.

File: pkg/commands/backend/create.go [hunk 3]
```diff
@@ -39,6 +40,7 @@ func NewCreateCommand(parent cmd.Registerer, globals *config.Data, data manifest
 	c.manifest = data
 	c.CmdClause = parent.Command("create", "Create a backend on a Fastly service version").Alias("add")
 	c.RegisterServiceIDFlag(&c.manifest.Flag.ServiceID)
+	c.RegisterServiceNameFlag(c.serviceName.Set, &c.serviceName.Value)
```
Comment (Integralist) [inline pkg/commands/backend/create.go:43]: Here is an example of using the new `--service-name` abstraction.

**Review by Integralist — COMMENTED** [review]
I've commented to help highlight the key changes...

**Review by kailan — APPROVED** [review]
This is a change I've been looking forward to and it was easy to review thanks to the comments 👍 

ship it!

**Pull request merged**